auth/proofOfPossession_test.go

```diff
@@ -0,0 +1,172 @@
+}
+
+func TestProofOfPossession_validateX5tThumbprint(t *testing.T) {
+	tests := []struct {
+		name              string
+		claimCnfMemberX5t string
+		certFile          string // in case of empty string it gets generated
+		pemEncoded        bool
+		expectedErrMsg    string // in case of empty string no error is expected
+	}{
+		{
+			name:              "x5t should match with DER certificate (go router)",
+			claimCnfMemberX5t: "fU-XoQlhMTpQsz9ArXl6zHIpMGuRO4ExLKdLRTc5VjM",
+			certFile:          "x-forwarded-client-cert.txt",
+			pemEncoded:        false,
+			expectedErrMsg:    "",
+		}, {
+			name:              "x5t should match with PEM certificate (apache proxy)",
+			claimCnfMemberX5t: "fU-XoQlhMTpQsz9ArXl6zHIpMGuRO4ExLKdLRTc5VjM",
+			certFile:          "x-forwarded-client-cert.txt",
+			pemEncoded:        true,
+			expectedErrMsg:    "",
+		}, {
+			name:              "expect error when x5t does not match with generated DER certificate (go router)",
+			claimCnfMemberX5t: "fU-XoQlhMTpQsz9ArXl6zHIpMGuRO4ExLKdLRTc5VjM",
+			certFile:          "",
+			pemEncoded:        false,
+			expectedErrMsg:    "token thumbprint confirmation failed",
+		}, {
+			name:              "expect error when x5t does not match with generated PEM certificate (apache proxy)",
+			claimCnfMemberX5t: "fU-XoQlhMTpQsz9ArXl6zHIpMGuRO4ExLKdLRTc5VjM",
+			certFile:          "",
+			pemEncoded:        true,
+			expectedErrMsg:    "token thumbprint confirmation failed",
+		}, {
+			name:              "expect error when x5t is empty",
+			claimCnfMemberX5t: "",
+			certFile:          "",
+			pemEncoded:        false,
+			expectedErrMsg:    "token provides no cnf member for thumbprint confirmation",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var cert string
+			if tt.certFile == "" {
+				cert = generateCert(t, tt.pemEncoded)
+			} else {
+				cert = readCert(t, tt.certFile, tt.pemEncoded)
+			}
+			x509cert, err := newCertificate(cert)
+			require.NoError(t, err, "Failed to validate client cert with token cnf thumbprint: %v", err)
+
+			err = ValidateX5tThumbprint(x509cert, createToken(t, tt.claimCnfMemberX5t))
+			if tt.expectedErrMsg != "" {
+				assert.Equal(t, tt.expectedErrMsg, err.Error())
+			} else {
+				require.NoError(t, err, "Failed to validate client cert with token cnf thumbprint: %v", err)
+			}
+		})
+	}
+}
+
+func createToken(t *testing.T, claimCnfMemberX5tValue string) Token {
+	token := jwt.New()
+	cnfClaim := map[string]interface{}{
+		claimCnfMemberX5t: claimCnfMemberX5tValue,
+	}
+	err := token.Set(claimCnf, cnfClaim)
+	require.NoError(t, err, "Failed to create token: %v", err)
+
+	return stdToken{jwtToken: token}
+}
+
+func readCert(t *testing.T, fileName string, pemEncoded bool) string {
+	pwd, _ := os.Getwd()
+	certFilePath := path.Join(pwd, "testdata", fileName)
+	certificate, err := os.ReadFile(certFilePath)
+	require.NoError(t, err, "Failed to read certificate from %v: %v", certFilePath, err)
```

Comment:
Relative paths work as well. Just use `"testdata" + fileName`.

Another pretty handy feature is go:embed: https://pkg.go.dev/embed
You can just define the following var outside the func and the content will be loaded automatically:

```go
//go:embed testdata/x-forwarded-client-cert.txt
var testCert string
```

Reply:
done